src/cluster_legacy.c

```diff
@@ -273,6 +283,54 @@ int auxHumanNodenamePresent(clusterNode *n) {
     return sdslen(n->human_nodename);
 }
 
+int auxClientIpV4Setter(clusterNode *n, void *value, int length) {
+    if (n && !strncmp(value, n->client_ipv4, length)) {
+        return C_OK;
+    } else if (!n && (length == 0)) {
+        return C_OK;
+    }
+    if (n) {
+        n->client_ipv4 = sdscpylen(n->client_ipv4, value, length);
+    } else if (sdslen(n->client_ipv4) != 0) {
+        sdsclear(n->client_ipv4);
+    } else {
+        return C_ERR;
+    }
+    return C_OK;
+}
+
+sds auxClientIpV4Getter(clusterNode *n, sds s) {
+    return sdscatprintf(s, "%s", n->client_ipv4);
+}
+
+int auxClientIpV4Present(clusterNode *n) {
+    return sdslen(n->client_ipv4) != 0;
+}
+
+int auxClientIpV6Setter(clusterNode *n, void *value, int length) {
+    if (n && !strncmp(value, n->client_ipv6, length)) {
+        return C_OK;
+    } else if (!n && (length == 0)) {
+        return C_OK;
+    }
+    if (n) {
+        n->client_ipv6 = sdscpylen(n->client_ipv6, value, length);
+    } else if (sdslen(n->client_ipv6) != 0) {
+        sdsclear(n->client_ipv6);
+    } else {
+        return C_ERR;
+    }
+    return C_OK;
+}
```

Comment:
Looks like this function is duplicated from `auxHumanNodenameSetter`, which unnecessarily and incorrectly assumes `n` could be NULL, while `n != NULL` is guaranteed by the contract already. Also I think it would be good to validate the input ipv6 address. 

```suggestion
int auxClientIpV6Setter(clusterNode *n, void *value, int length) {
    struct sockaddr_in6 sa;

    /* Validate IPv6 address */
    if (inet_pton(AF_INET6, (const char *)value, &(sa.sin6_addr)) == 0) {
        return C_ERR; /* Return error if the value is not a valid IPv6 address */
    }

    /* Check if the new value matches the current value */
    if (!strncmp((const char *)value, n->client_ipv6, length)) {
        return C_OK; /* No change needed if the new value matches the current value */
    }
    
    /* Handle length 0 case */
    if (length == 0) {
        if (sdslen(n->client_ipv6) != 0) {
            sdsclear(n->client_ipv6); /* Clear the current value if the length is 0 */
        }
    } else {
        /* Copy the new value */
        n->client_ipv6 = sdscpylen(n->client_ipv6, (const char *)value, length);
    }
    
    return C_OK;
}
```

Reply:
Yes, I copied either hostname or human nodename. :hand_over_mouth: 

Validation is good. I'll do that in the setter in config.c too.

Some comments on your suggestion though:

* We don't want to validate when setting to the empty string.
* `!strncmp((const char *)value, n->client_ipv6, length)` just checks if `n->client_ipv6` is a prefix of the new value. It's not an equality check. Need to compare sdslen to length also.
* No need to use sdsclear. We can use sdscpylen also for length == 0. It has the same effect.

I believe these are flaws in the human nodename code as well. I can fix it in a separate PR.